core/server/common/src/main/java/alluxio/RpcUtils.java

```diff
@@ -39,10 +39,35 @@
     try {
       return callable.call();
     } catch (AlluxioException e) {
-      LOG.debug("Internal Alluxio error when running rpc", e);
+      LOG.debug("Internal Error: {}", callable, e);
       throw e.toThrift();
     } catch (Exception e) {
-      LOG.error("Unexpected error running rpc", e);
+      LOG.error("Unexpected Error: {}", callable, e);
+      throw new UnexpectedAlluxioException(e).toThrift();
+    }
+  }
+
+  /**
+   * Calls the given {@link RpcCallable} and handles any exceptions thrown. This method also logs
+   * enter and exit when debug level logging is enabled.
+   *
+   * @param logger the logger to use for this call
+   * @param callable the callable to call
+   * @param <T> the return type of the callable
+   * @return the return value from calling the callable
+   * @throws AlluxioTException if the callable throws an Alluxio or runtime exception
+   */
+  public static <T> T call(Logger logger, RpcCallable<T> callable) throws AlluxioTException {
+    try {
+      logger.debug("Enter: {}", callable);
+      T ret = callable.call();
+      logger.debug("Exit: {}", callable);
```

Comment:
I like the `Exit (Error Type)` format, since we are explicitly categorizing the errors, I think it is valuable to propagate this to the user.